openmc/statepoint.py

```diff
@@ -389,7 +389,7 @@ def tallies(self):
                         new_filter.mesh = self.meshes[key]
 
                     # Add Filter to the Tally
-                    tally.add_filter(new_filter)
+                    tally.filters.append(new_filter)
```

Comment:
Looks like you wrote this while I was typing up my final comment below. I think our thought processes are converging here and will let you decide what's best.